### ISSUE TITLE
a number of exceptions (trying sample quit process, etc) thrown while trying austin-tui -C -m -l on shortlived process

### ISSUE THREAD

**Issue by yarikoptic** [P403n1x87/austin#51]
<details>
<summary>version information</summary> 

austin itself installed from debian

```shell
$> which austin ; apt-cache policy austin
/usr/bin/austin
austin:
  Installed: 1.0.1-2
  Candidate: 1.0.1-2
  Version table:
 *** 1.0.1-2 600
        600 http://http.debian.net/debian sid/main amd64 Packages
        100 /var/lib/dpkg/status
     1.0.0-1 900
        900 http://deb.debian.org/debian bullseye/main amd64 Packages
```
but it is a bare C tool without any python tui, etc.  So I cloned this repo at current state of

```
$> git describe
fatal: No annotated tags can describe '679a89676c3b5cb17a1766a9e5c4a3f335ddc495'.
However, there were unannotated tags: try --tags.

$> git describe --tags
v1.0.1-1-g679a896
```

</details>

I am trying austin-tui in a virtualenv into which I also installed datalad, which I'd like to profile... commands were 
```
(git)lena:~/proj/misc/austin[master]git
$> py=3; d=venvs/dev$py; virtualenv --python=python$py --system-site-packages $d && source $d/bin/activate && pip3 install -e .
$> pip install datalad   
```

running
```
$> venvs/dev3/bin/austin-tui -C -m -l -- /home/yoh/proj/misc/austin/venvs/dev3/bin/datalad --help
```
results in nothing really visible in tui (not surprising -- short lived process) and coming back to terminal with following tracebacks
```
$> venvs/dev3/bin/austin-tui -C -m -l -- /home/yoh/proj/misc/austin/venvs/dev3/bin/datalad --help
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/psutil/_common.py", line 340, in wrapper
    ret = self._cache[fun]
AttributeError: _cache

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/psutil/_pslinux.py", line 1517, in wrapper
    return fun(self, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/psutil/_common.py", line 343, in wrapper
    return fun(self)
  File "/usr/lib/python3/dist-packages/psutil/_pslinux.py", line 1559, in _parse_stat_file
    with open_binary("%s/%s/stat" % (self._procfs_path, self.pid)) as f:
  File "/usr/lib/python3/dist-packages/psutil/_common.py", line 604, in open_binary
    return open(fname, "rb", **kwargs)
FileNotFoundError: [Errno 2] No such file or directory: '/proc/966126/stat'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/yoh/proj/misc/austin/austin/tui.py", line 596, in start
    self.run(self._scr)
  File "/home/yoh/proj/misc/austin/austin/tui.py", line 575, in run
    task.result()
  File "/home/yoh/proj/misc/austin/austin/tui.py", line 558, in update_loop
    self.get_child("cpu_plot").push(self.get_current_cpu())
  File "/home/yoh/proj/misc/austin/austin/tui.py", line 388, in get_current_cpu
    value = int(self.austin.get_child().cpu_percent())
  File "/usr/lib/python3/dist-packages/psutil/__init__.py", line 1112, in cpu_percent
    pt2 = self._proc.cpu_times()
  File "/usr/lib/python3/dist-packages/psutil/_pslinux.py", line 1517, in wrapper
    return fun(self, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/psutil/_pslinux.py", line 1710, in cpu_times
    values = self._parse_stat_file()
  File "/usr/lib/python3/dist-packages/psutil/_pslinux.py", line 1524, in wrapper
    raise NoSuchProcess(self.pid, self._name)
psutil.NoSuchProcess: psutil.NoSuchProcess process no longer exists (pid=966126)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "venvs/dev3/bin/austin-tui", line 11, in <module>
    load_entry_point('python-austin', 'console_scripts', 'austin-tui')()
  File "/home/yoh/proj/misc/austin/austin/main.py", line 23, in main
    curses.wrapper(lambda scr: curses_app(scr, parsed_args))
  File "/usr/lib/python3.8/curses/__init__.py", line 105, in wrapper
    return func(stdscr, *args, **kwds)
  File "/home/yoh/proj/misc/austin/austin/main.py", line 23, in <lambda>
    curses.wrapper(lambda scr: curses_app(scr, parsed_args))
  File "/home/yoh/proj/misc/austin/austin/main.py", line 10, in curses_app
    austin_tui.start(args)
  File "/home/yoh/proj/misc/austin/austin/tui.py", line 600, in start
    self.austin.join()
  File "/home/yoh/proj/misc/austin/austin/__init__.py", line 274, in join
    return self._loop.run_until_complete(self._start_task)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/yoh/proj/misc/austin/austin/__init__.py", line 236, in _start
    self._callback(data.decode("ascii").rstrip())
  File "/home/yoh/proj/misc/austin/austin/tui.py", line 349, in on_sample_received
    self.stats.add_thread_sample(line.encode())
  File "/home/yoh/proj/misc/austin/austin/stats.py", line 13, in atomic_wrapper
    result = f(*args, **kwargs)
  File "/home/yoh/proj/misc/austin/austin/stats.py", line 134, in add_thread_sample
    sample_stack = SampledFrame(frames, duration, 1) if frames else None
  File "/home/yoh/proj/misc/austin/austin/stats.py", line 34, in __init__
    self.children = [SampledFrame(data[2:], duration, height + 1)]
  File "/home/yoh/proj/misc/austin/austin/stats.py", line 34, in __init__
    self.children = [SampledFrame(data[2:], duration, height + 1)]
  File "/home/yoh/proj/misc/austin/austin/stats.py", line 34, in __init__
    self.children = [SampledFrame(data[2:], duration, height + 1)]
  [Previous line repeated 66 more times]
  File "/home/yoh/proj/misc/austin/austin/stats.py", line 31, in __init__
    self.line_number = data[1][1:]
IndexError: list index out of range
```

**Comment by yarikoptic** [comment]
I now even rebuilt `austin` binary straight from sources (instead of using debian pkg) while trying to get more info for #52 (which has a "full" reproducibility script) , and went into epdb to see what `data` looked like there and it was just 
```
(Epdb) p data
['L1483']
```
going up in the stack and showing more info about what it parsed/processes;
```
> /home/yoh/proj/misc/austin/austin/__init__.py(236)_start()
-> self._callback(data.decode("ascii").rstrip())
(Epdb) p data
b'Process 1006282;Process 1006282;Process 1006282;Process 1006282;Process 1006282;Process 1006282;Thread 7ff3c99b0740;<module> (/usr/bin/datalad);L7;_find_and_load (<frozen importlib._bootstrap>);L991;_find_and_load_unlocked (<frozen importlib._bootstrap>);L961;_call_with_frames_removed (<frozen importlib._bootstrap>);L219;_find_and_load (<frozen importlib._bootstrap>);L991;_find_and_load_unlocked (<frozen importlib._bootstrap>);L961;_call_with_frames_removed (<frozen importlib._bootstrap>);L219;_find_and_load (<frozen importlib._bootstrap>);L991;_find_and_load_unlocked (<frozen importlib._bootstrap>);L971;_find_spec (<frozen importlib._bootstrap>);L914;find_spec (<frozen importlib._bootstrap_external>);L1342;_get_spec (<frozen importlib._bootstrap_external>);L1314;find_spec (<frozen importlib._bootstrap_external>);L1443;_fill_cache (<frozen importlib._bootstrap_external>);L1483 264\n'
(Epdb) down
> /home/yoh/proj/misc/austin/austin/tui.py(349)on_sample_received()
-> self.stats.add_thread_sample(line.encode())
(Epdb) p line.encode()
b'Process 1006282;Process 1006282;Process 1006282;Process 1006282;Process 1006282;Process 1006282;Thread 7ff3c99b0740;<module> (/usr/bin/datalad);L7;_find_and_load (<frozen importlib._bootstrap>);L991;_find_and_load_unlocked (<frozen importlib._bootstrap>);L961;_call_with_frames_removed (<frozen importlib._bootstrap>);L219;_find_and_load (<frozen importlib._bootstrap>);L991;_find_and_load_unlocked (<frozen importlib._bootstrap>);L961;_call_with_frames_removed (<frozen importlib._bootstrap>);L219;_find_and_load (<frozen importlib._bootstrap>);L991;_find_and_load_unlocked (<frozen importlib._bootstrap>);L971;_find_spec (<frozen importlib._bootstrap>);L914;find_spec (<frozen importlib._bootstrap_external>);L1342;_get_spec (<frozen importlib._bootstrap_external>);L1314;find_spec (<frozen importlib._bootstrap_external>);L1443;_fill_cache (<frozen importlib._bootstrap_external>);L1483 264'
(Epdb) down
> /home/yoh/proj/misc/austin/austin/stats.py(13)atomic_wrapper()
-> result = f(*args, **kwargs)
(Epdb) 
> /home/yoh/proj/misc/austin/austin/stats.py(138)add_thread_sample()
-> sample_stack = SampledFrame(frames, duration, 1) if frames else None
(Epdb) p frames
['Process 1006282', 'Process 1006282', 'Process 1006282', 'Process 1006282', 'Thread 7ff3c99b0740', '<module> (/usr/bin/datalad)', 'L7', '_find_and_load (<frozen importlib._bootstrap>)', 'L991', '_find_and_load_unlocked (<frozen importlib._bootstrap>)', 'L961', '_call_with_frames_removed (<frozen importlib._bootstrap>)', 'L219', '_find_and_load (<frozen importlib._bootstrap>)', 'L991', '_find_and_load_unlocked (<frozen importlib._bootstrap>)', 'L961', '_call_with_frames_removed (<frozen importlib._bootstrap>)', 'L219', '_find_and_load (<frozen importlib._bootstrap>)', 'L991', '_find_and_load_unlocked (<frozen importlib._bootstrap>)', 'L971', '_find_spec (<frozen importlib._bootstrap>)', 'L914', 'find_spec (<frozen importlib._bootstrap_external>)', 'L1342', '_get_spec (<frozen importlib._bootstrap_external>)', 'L1314', 'find_spec (<frozen importlib._bootstrap_external>)', 'L1443', '_fill_cache (<frozen importlib._bootstrap_external>)', 'L1483']
```

**Comment by P403n1x87** [comment]
Could you try with the command (no `--`)

~~~ bash
$> venvs/dev3/bin/austin-tui -Cml /home/yoh/proj/misc/austin/venvs/dev3/bin/datalad --help
~~~

The `-l` *might* be broken so, if that fails, try without it

~~~ bash
$> venvs/dev3/bin/austin-tui -Cm /home/yoh/proj/misc/austin/venvs/dev3/bin/datalad --help
~~~

In the TUI itself, press `F` to display the full collected statistics.

**Comment by yarikoptic** [comment]
without `--` you (on any python script I assume) would get `--help` output of austin-tui, doesn't it do that for you?

```shell
$> venvs/dev3/bin/austin-tui -Cml /home/yoh/proj/misc/austin/venvs/dev3/bin/datalad --help
usage: austin-tui [-h] [-C] [-e] [-i INTERVAL] [-m] [-p PID] [-s] [-t TIMEOUT] [-l] [command] [args [args ...]]

positional arguments:
  command               The command to execute if no PID is provided, followed by its arguments.
  args                  Arguments to pass to the command to run.

optional arguments:
  -h, --help            show this help message and exit
  -C, --children        Attach to child processes.
  -e, --exclude-empty   Do not output samples of threads with no frame stacks.
  -i INTERVAL, --interval INTERVAL
                        Sampling interval (default is 100us).
  -m, --memory          Profile memory usage.
  -p PID, --pid PID     The the ID of the process to which Austin should attach.
  -s, --sleepless       Suppress idle samples.
  -t TIMEOUT, --timeout TIMEOUT
                        Approximate start up wait time. Increase on slow machines (default is 100ms).
  -l, --linenos         Show line numbers
```
`--` is a convention in many tools/arg parsers to separate out "wrapped command" options from the tool's options.  So it is in the argparse: https://docs.python.org/3/library/argparse.html 

If you have positional arguments that must begin with - and don’t look like negative numbers, you can insert the pseudo-argument '--' which tells parse_args() that everything after that is a positional argument:
```
>>>
>>> parser.parse_args(['--', '-f'])
Namespace(foo='-f', one=None)
```

**Comment by P403n1x87** [comment]
Yes you are right :). For a proper project, this would be a way of handling this scenario. The TUI that comes with Austin is more of an example than a properly maintained tool. This problem will be fixed with the release of a new version of the TUI, which will be its own project, when Austin 2 comes out in a few months with Python 3.9 support.

**Comment by P403n1x87** [comment]
Fixed by [austin-tui@v0.1.2](https://github.com/P403n1x87/austin-tui/releases/tag/v0.1.2).
